py3status/modules/wifi.py

```diff
@@ -89,15 +89,13 @@ def post_config_hook(self):
             raise Exception(STRING_NOT_INSTALLED)
 
         # get wireless interface
-        try:
+        if not self.device:
             data = self.py3.command_output([iw, "dev"])
-            devices = re.findall(r"Interface\s*([^\s]+)", data)
-            if not devices or "wlan0" in devices:
-                self.device = "wlan0"
-            else:
-                self.device = devices[0]
-        except self.py3.CommandError:
-            pass
+            for line in data.splitlines()[1:]:
+                if "Interface" in line:
+                    self.device = line.split()[-1]
```

Comment:
It looks like previously we prioritized `wlan0` if such device was present in any position, now we are just taking the first available one?

Reply:
Defaulting `wlan0` does not make sense. Leave it `None` so we can detect the first `device`. If users specified `device`, we won't do anything but to just try it with that `device`.

This PR clean up/simplify the code to stop defaulting `device` with `wlan0`.